docs/rule/no-autofocus-attribute.md

```diff
@@ -0,0 +1,30 @@
+# no-autofocus-attribute
+
+🔧 The `--fix` option on the command line can automatically fix some of the problems reported by this rule.
+
+Enforce no autofocus prop on element. The autofocus attribute is a global attribute that indicates an element should be focused on page load. Because autofocus reduces accessibility by moving users to an element without warning and context, it should not be used.
+
+This rule takes no arguments.
+
+## Examples
+
+This rule **allows** the following:
```

Comment:
I've used `autofocus` to get focus onto a central search textbox on a web page upon page load so the user can quickly begin typing, just like how google.com uses it on its big search box. I assume this is a very common pattern, so we should probably mention use this use case and offer suggestions for it?

Reply:
Follow-up question, due to this use case, would this rule ever be suitable to be `recommended`? CC: @rwjblue

Reply:
disallowing autofocus should be a recommended rule.

Reply:
@MelSumner I see. For the above use case I mentioned, do we just recommend people inline disable the rule for that?

Reply:
Yes, that's the recommendation for all A11y rules that might have an exception that the author feels is valid.